storage/engine.go

```diff
@@ -531,6 +531,11 @@ func (e *Engine) DeleteBucket(orgID, bucketID platform.ID) error {
 
 // DeleteBucketRange deletes an entire bucket from the storage engine.
 func (e *Engine) DeleteBucketRange(orgID, bucketID platform.ID, min, max int64) error {
+	// Snapshot to clear the cache to reduce write contention.
+	if err := e.engine.WriteSnapshot(context.Background()); err != nil {
+		return err
+	}
+
```

Comment:
Do we need to worry about multiple deletes causing too many snapshots? Not sure how expensive it is to write snapshots.

Reply:
This is in the "delete a whole range from a bucket" code path called by retention and bucket deletion, so we shouldn't have to worry about that. We definitely don't want to call snapshot on every delete as it is expensive, but the expensive part of the snapshot at least won't block incoming writes.